File: java/po/world_simulator/entities/animals/Wolf.java
package po.world_simulator.entities.animals;

import java.util.Map;

import po.world_simulator.Vector2;
import po.world_simulator.entities.Animal;

public class Wolf extends Animal {
    public Wolf(Vector2 position, int power, int age) {
        super(power, 5, position, "Wolf", age, true);
    }

    public Wolf(Vector2 position) {
        this(position, 9, 0);
    }

    public Wolf(Map<String, String> entityData){
        this(
            new Vector2(entityData.get("position")),
            Integer.parseInt(entityData.get("power")),
            Integer.parseInt(entityData.get("age"))
        );
    }
}

File: python/po/world_simulator/entities/animals/Wolf.py
from po.world_simulator.entities.Animal import Animal
from po.world_simulator.Vector2 import Vector2

class Wolf(Animal):
    # [AI] Merged constructors.
    def __init__(self, position_or_data, power=9, age=0):
        if isinstance(position_or_data, dict):
            super().__init__(
                int(position_or_data.get("power", 0)),
                5,
                Vector2(position_or_data.get("position")),
                "Wolf",
                int(position_or_data.get("age", 0)),
                True
            )
        else:
            super().__init__(power, 5, position_or_data, "Wolf", age, True)